api/jobs.go

```diff
@@ -24,84 +29,125 @@ var (
 	serverErrorMessage = "internal server error"
 )
 
-// CreateJobHandler returns a function that generates a new Job resource containing default values in its fields.
-func (api *API) CreateJobHandler(ctx context.Context) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-		ctx := req.Context()
+// CreateJobHandler generates a new Job resource and a new ElasticSearch index associated with it	.
+func (api *API) CreateJobHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	id := NewID()
 
-		id := NewID()
+	log.Info(ctx, "creating new job resource in the data store")
+	newJob, err := api.dataStore.CreateJob(ctx, id)
+	if err != nil {
+		log.Error(ctx, "creating and storing job failed", err)
+		if err == mongo.ErrExistingJobInProgress {
+			http.Error(w, "existing reindex job in progress", http.StatusConflict)
+		} else {
+			http.Error(w, serverErrorMessage, http.StatusInternalServerError)
+		}
+		return
+	}
 
-		newJob, err := api.dataStore.CreateJob(ctx, id)
-		if err != nil {
-			log.Error(ctx, "creating and storing job failed", err)
-			if err == mongo.ErrExistingJobInProgress {
-				http.Error(w, "existing reindex job in progress", http.StatusConflict)
-			} else {
+	log.Info(ctx, "creating new index in ElasticSearch via the Search API")
+	serviceAuthToken := "Bearer " + api.cfg.ServiceAuthToken
+	searchAPISearchURL := api.cfg.SearchAPIURL + "/search"
```

Comment:
do we have a rule/utility protecting against the possibility that SearchAPIURL already terminates with a `/` (or do we just ignore double slashes?)

Reply:
maybe next time